README.rst

```diff
@@ -340,6 +340,57 @@ unconventional syntax (without the usual ``@``):
     if isinstance(res, Ok):
         print(res.value)
 
+
+Do notation: syntactic sugar for a sequence of `and_then()` calls.
+Much like the equivalent in Rust or Haskell, but with different syntax.
+Instead of `x <- Ok(1)` we write `for x in Ok(1)`.
+Since the syntax is generator-based, the final result must be the first line,
+not the last.
+
+.. sourcecode:: python
+
+
+    >>> final_result: Result[float, int] = do(
+            Ok(len(x) + int(y) + 0.5)
+            for x in Ok("hello")
+            for y in Ok(True)
+        )
+
+    >>> # This is similar to
+    use do_notation::m;
+    let final_result = m! {
+        x <- Ok("hello");
+        y <- Ok(True);
+        Ok(len(x) + int(y) + 0.5)
+    };
+
+Async functions with do notation: no special considerations.
```

Comment:
Could we reword this, perhaps something like, `Async functions can also be used:`